app/managers/performance_metrics_manager/index.ts

```diff
@@ -44,17 +49,32 @@ class PerformanceMetricsManager {
     }
 
     public finishLoad(location: Target, serverUrl: string) {
+        this.finishLoadWithRetries(location, serverUrl, 0);
+    }
+
+    private finishLoadWithRetries(location: Target, serverUrl: string, retries: number) {
         if (this.target !== location || this.hasRegisteredLoad) {
             return;
         }
 
-        const measure = performance.measure('mobile_load', 'nativeLaunchStart');
-        this.ensureBatcher(serverUrl).addToBatch({
-            metric: 'mobile_load',
-            value: measure.duration,
-            timestamp: Date.now(),
-        });
-        performance.clearMeasures('mobile_load');
+        try {
+            const measure = performance.measure('mobile_load', 'nativeLaunchStart');
+            this.ensureBatcher(serverUrl).addToBatch({
+                metric: 'mobile_load',
+                value: measure.duration,
+                timestamp: Date.now(),
+            });
+            performance.clearMeasures('mobile_load');
+        } catch {
+            // There seems to be a race condition where in some scenarios, the mobile load
+            // mark does not exist. We add this to avoid crashes related to this.
+            if (retries < MAX_RETRIES) {
+                setTimeout(() => this.finishLoadWithRetries(location, serverUrl, retries + 1), RETRY_TIME);
+                return;
+            }
+            logWarning('We could not retrieve the mobile load metric');
+        }
```

Comment:
Q: if in the event it fails 3x, should we change `this.hasRegisteredLoad` to `true`?

Reply:
Yes, we should. `hasRegisteredLoad` is used simply so we don't send again the load metric when any of the "load points" is reached (e.g. channel list is loaded).

If we don't set it, this may trigger again a few seconds or even minutes later, and then we may have the value. But by that point, the metric we would store is completely off. Therefore, we prefer to lose one data point.